### PR TITLE
Add references to launch templates from EKS node groups

### LINKED ISSUE #851: Need to enable launch template ref from EKS node groups
### What problem are you facing?
EKS node groups do not have ref fields for launch templates requiring data to be passed via the composite.

### How could Official AWS Provider help solve your problem?
I will contribute a PR to add the refs.

### PR DESCRIPTION
**Pull request by oblogic7**
### Description of your changes

Fixes #851

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Guidance needed.

### CODE REVIEW

**Comment by oblogic7** [comment]
I have read the crossplane and upjet contribution guides.  I'm still not certain of the most efficient path for testing changes to a provider.  Specifically, is there tooling or commands available that offer an efficient path for standing up a local dev environment for testing provider changes?

**Comment by turkenf** [comment]
/test-examples="examples/eks/nodegroup.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/eks/nodegroup.yaml"

**Comment by turkenf** [comment]
@oblogic7 uptest run failed due to the dependent resources did not exist in the example manifest, we should test manually.

File: apis/eks/v1beta1/zz_nodegroup_types.go [hunk 1]
```diff
@@ -27,12 +27,6 @@ type AutoscalingGroupsParameters struct {
 
 type LaunchTemplateInitParameters struct {
 
-	// Identifier of the EC2 Launch Template. Conflicts with name.
-	ID *string `json:"id,omitempty" tf:"id,omitempty"`
-
-	// Name of the EC2 Launch Template. Conflicts with id.
-	Name *string `json:"name,omitempty" tf:"name,omitempty"`
-
```
Comment (turkenf) [inline apis/eks/v1beta1/zz_nodegroup_types.go:35]: @lsviben, Is it expected to be removed from InitParameters when we add a reference to an argument?

File: config/eks/config.go [hunk 2]
```diff
@@ -52,6 +52,16 @@ func Configure(p *config.Provider) { // nolint:gocyclo
 			RefFieldName:      "SubnetIDRefs",
 			SelectorFieldName: "SubnetIDSelector",
 		}
+		r.References["launch_template.id"] = config.Reference{
+			Type:              "github.com/upbound/provider-aws/apis/ec2/v1beta1.LaunchTemplate",
+			RefFieldName:      "LaunchTemplateIDRefs",
+			SelectorFieldName: "LaunchTemplateIDSelector",
```
Comment (turkenf) [inline config/eks/config.go:58]:
```suggestion
			Type:              "github.com/upbound/provider-aws/apis/ec2/v1beta1.LaunchTemplate",
			TerraformName: "aws_kinesis_stream",
			Extractor:     common.PathTerraformIDExtractor,
```

The better way to use `TerraformName` even if it is difficult, in order not to be affected by any group changes. 
And we'd better not use `RefFieldName` and `SelectorFieldName` unless you have a specific reason to come up with it that way. It is used in the configuration you added as follows:
```
spec:
  launchTemplate:
    launchTemplateIdSelector:
      matchLabels:
        testing.upbound.io/example-name: ....
```
But as recommended:
```
spec:
  launchTemplate:
    IDSelector:
      matchLabels:
        testing.upbound.io/example-name: ....
```
What do you think?

File: config/eks/config.go [hunk 3]
```diff
@@ -52,6 +52,16 @@ func Configure(p *config.Provider) { // nolint:gocyclo
...
+		}
+		r.References["launch_template.name"] = config.Reference{
+			Type:              "github.com/upbound/provider-aws/apis/ec2/v1beta1.LaunchTemplate",
+			RefFieldName:      "LaunchTemplateNameRefs",
+			SelectorFieldName: "LaunchTemplateNameSelector",
+		}
```
Comment (turkenf) [inline config/eks/config.go:64]: We may not add this as name and id cannot be used together.

File: apis/eks/v1beta1/zz_nodegroup_types.go (hunk 1 above)
Comment (lsviben) [reply apis/eks/v1beta1/zz_nodegroup_types.go:35]: yes, references are not a part of initProvider, ref: https://github.com/upbound/upjet/blob/14cd59c9986d2a6bc07ea8c0282cde4cad504644/pkg/types/field.go#L282

Comment (turkenf) [reply apis/eks/v1beta1/zz_nodegroup_types.go:35]: Thanks for the information.

**Comment by turkenf** [comment]
@oblogic7, if you are considering taking this PR forward, please check the comments.

**Comment by turkenf** [comment]
I am closing this now if you want to take forward feel free to reopen it.

**Pull request closed without merging**